warehouse/config.py

```diff
@@ -63,6 +63,13 @@ class RootFactory:
         (Allow, "group:psf_staff", "psf_staff"),
         (Allow, "group:with_admin_dashboard_access", "admin_dashboard_access"),
         (Allow, Authenticated, "manage:user"),
+        # OIDC beta users can see their user-level OIDC settings.
+        # XXX: Can be removed once OIDC is removed from beta.
+        (
+            Allow,
+            "group:oidc_beta",
+            "manage:user:oidc",
+        ),
```

Comment:
Flagging: this isn't 100% right, since it gives everybody in the OIDC beta the ability to modify every other OIDC beta users' OIDC providers. So maybe a principal is the wrong abstraction here.